feat: add support for associating tags with a feature flag

weyert opened the pull request:
## Changes

Updates the Add Feature Flag page to allow associating one or more tags
with a feature flag

## How did you test this code?

I have manually executed the following steps:

  - Add tag, and save feature flag
  - Reload feature flag
  - Remove existing tag and save feature flag

tapico-weyert committed:
feat: add support for associating tags with a feature flag

Updates the Add Feature Flag page to allow associating one or more tags
with a feature flag

weyert commented:
Screenshot:

<img width="740" alt="CleanShot 2021-12-01 at 23 39 19@2x" src="https://user-images.githubusercontent.com/7049/144331646-fdc3ea4d-b8e7-43e7-b9e8-acf932ec13a6.png">

tapico-weyert committed:
test: add end-to-end test for feature flag tags

weyert commented:
@paolodamico @macobo @timgl I have made a PR to introduce basic support for associating tags with feature flag. Let me know, what you think of it.

My main open question how would filtering by tags work for on the feature flags overview page?

neilkakkar commented:
Hey Weyert! Thanks for PR-ing! I'm a bit out of the loop: What's the use-case for tags here? Did you face a problem that's being solved by these?

weyert commented:
Mainly organising of the feature flags; so can filter them by tag, e.g. like `ops`, `development`, `experiment` tags :)

paolodamico commented:
Thanks for putting this out @weyert, I think it'll make it easier to keep things organized. For consistency, filtering should look the same way we have it for saved insights (hope this answers your question). Before continuing with this, I do need to check internally how this fits with all the changes we're making to experimentation, hope that's okay.

<img width="1187" alt="" src="https://user-images.githubusercontent.com/5864173/144492093-2d3e6bce-d4d8-47eb-acec-48fa0b3555a0.png">

paolodamico commented:
Also just friendly reminder @weyert to follow the same pattern as we have on Dashboards for tags regarding the feature structure and requiring a paid license to use.

tapico-weyert committed:
chore: Merge branch 'master' into allow-tags-for-feature-flag

tapico-weyert committed:
feat: add `/api/projects/:team/feature_flags/tags` endpoint

Introducing a new endpoint to fetch all tags used by all the defined
feature flags

tapico-weyert committed:
feat: add filtering by tag in the Flags overview page

Allow to filter out feature flags by a chosen tag

Added feature flag named `feature-flags-tags` when ebaled the `tags` UI
elements will be available

weyert commented:
Added the ability to filter the feature flags by a selected tag and  put the functionality behind a feature flag named `feature-flag-tags` :)

![image](https://user-images.githubusercontent.com/7049/144896565-0019701a-814a-4e64-b3ae-cf2aabf26c8a.gif)
BC1/cleanshot_2021-12-06_at_17.14.19.gif

I haven't been bale to reproduce the filter bar from your screenshot yet but I will revisit that later. If anyone have any pointers were to look.

tapico-weyert committed:
fix: add space around the button and filter component in the toolbar

neilkakkar commented on frontend/src/scenes/feature-flags/featureFlagsLogic.ts:
```diff
@@ -72,6 +96,7 @@ export const featureFlagsLogic = kea<featureFlagsLogicType>({
     events: ({ actions }) => ({
         afterMount: () => {
             actions.loadFeatureFlags()
+            actions.loadFeatureFlagTags()
```
this needs the FeatureFlag enabled check as well

neilkakkar commented on frontend/src/scenes/feature-flags/featureFlagLogic.tsx:
```diff
@@ -290,6 +293,26 @@ export const featureFlagLogic = kea<featureFlagLogicType>({
                 actions.setMultivariateOptions(null)
             }
         },
+        saveNewTag: ({ tag }) => {
```
think `featureFlagTags` should listen on this as well? Add this tag to the list when created

tapico-weyert committed:
chore: WIP commit my kea logic code snippets

weyert commented:
@posthog-bot  still working on it; just on vacation :)
